futures-core/src/future/mod.rs

```diff
@@ -139,11 +139,11 @@ impl<'a, F: ?Sized + Future> Future for PinMut<'a, F> {
 if_std! {
     use std::boxed::{Box, PinBox};
 
-    impl<'a, F: ?Sized + Future> Future for Box<F> {
+    impl<'a, F: ?Sized + Future + Unpin> Future for Box<F> {
```

Comment:
Note that I changed the `Box` impls not to project-- that allows us to be more conservative while we shake out whether or not we want `Box` to unconditionally be `Unpin`.

Reply:
(personally I'm still not sold on that choice, but I don't think we should take a hard soundness dependency upon unresolved decisions like this)

Reply:
Good catch.